python/tskit/trees.py

```diff
@@ -2116,15 +2116,45 @@ def num_tracked_samples(self, u=None):
         u = self.virtual_root if u is None else u
         return self._ll_tree.get_num_tracked_samples(u)
 
-    # TODO document these traversal arrays
-    # https://github.com/tskit-dev/tskit/issues/1788
     def preorder(self, u=NULL):
+        """
+        Returns a numpy array of node ids. Starting at `u`, yield the current node,
+        then recurse and do a preorder on each child of the current node. See also
+        `Wikipedia<https://en.wikipedia.org/wiki/Tree_traversal#Pre-order_(NLR)>`__.
+        Also see :ref:`tutorials:sec_analysing_trees_traversals` for examples of how
+        to use traversals.
+
+        :param int u: The node to start from, defaults to the virtual root.
+        :return: Array of node ids
+        :rtype: numpy.array[int32]
+        """
         return self._ll_tree.get_preorder(u)
 
     def postorder(self, u=NULL):
+        """
+        Returns a numpy array of node ids. Starting at `u`, recurse and do a postorder
+        on each child of the current node, then yield the current node. See also
+        `Wikipedia <https://en.wikipedia.org/wiki/Tree_traversal#Post-order_(LRN)>`__.
+        Also see :ref:`tutorials:sec_analysing_trees_traversals` for examples of how
+        to use traversals.
+
+        :param int u: The node to start from, defaults to the virtual root.
+        :return: Array of node ids
+        :rtype: numpy.array[int32]
+        """
         return self._ll_tree.get_postorder(u)
 
     def timeasc(self, u=NULL):
+        """
+        Returns a numpy array of node ids. Starting at `u`, visits the nodes in order of
+        increasing time, falling back to increasing ID if times are equal.
```

Comment:
Should we say "increasing time (youngest first)" so it's clear what direction time goes in?

Reply:
Done